Automattic/woocommerce-payments
# Add button to download deposits as CSV

Fixes #2548

This issue is similar to #689. The PR itself is heavily inspired by #1479.

#### Changes proposed in this Pull Request

Similar to how we have a download button on the Transactions page, these changes will add a download button on the Deposits page. This came up as an idea on the community forums and looked like a quick win as we would mostly need to just replicate the work done previously for transactions.

This is how the brand-new Download button will look like on the Deposits page:

<img width="1087" alt="image" src="https://user-images.githubusercontent.com/1288616/130308640-deed93e8-1b4c-4b88-bb18-87c11cf3cdf0.png">

Clicking the button will also record the action as a WC Tracks event.

Apart from these changes, I propose the following:

**Move shared styles to a common place.** Currently, the styles for this button are stored inside `client/transactions/list/style.scss`. I think the ideal new location for the styles would be `client/style.scss`.

#### Testing instructions

* Visit **Payments** → **Deposits**.
* Click Download button in the right upper corner of table.
* Adjust filters, sorting, and column display, and click Download again to ensure the changes are reflected in the exported CSV.

-------------------

- [ ] Event registered in Tracks ( PCYsg-r6P-p2 )
